projection/account_raw_event/account_raw_event.go

```diff
@@ -0,0 +1,530 @@
+	"github.com/crypto-com/chain-indexing/appinterface/rdb"
+	event_entity "github.com/crypto-com/chain-indexing/entity/event"
+	"github.com/crypto-com/chain-indexing/entity/projection"
+	applogger "github.com/crypto-com/chain-indexing/external/logger"
+	"github.com/crypto-com/chain-indexing/external/tmcosmosutils"
+	"github.com/crypto-com/chain-indexing/infrastructure/pg/migrationhelper"
+	"github.com/crypto-com/chain-indexing/projection/account_raw_event/view"
+	event_usecase "github.com/crypto-com/chain-indexing/usecase/event"
+	"github.com/crypto-com/chain-indexing/usecase/parser/utils"
+)
+
+var _ projection.Projection = &AccountRawEvent{}
+
+var (
+	NewAccountRawEvents          = view.NewAccountRawEventsView
+	NewAccountRawEventsTotal     = view.NewAccountRawEventsTotalView
+	UpdateLastHandledEventHeight = (*AccountRawEvent).UpdateLastHandledEventHeight
+)
+
+type AccountRawEvent struct {
+	*rdbprojectionbase.Base
+
+	accountAddressPrefix string
+
+	rdbConn rdb.Conn
+	logger  applogger.Logger
+
+	migrationHelper migrationhelper.MigrationHelper
+}
+
+func NewAccountRawEvent(
+	logger applogger.Logger,
+	rdbConn rdb.Conn,
+	accountAddressPrefix string,
+	migrationHelper migrationhelper.MigrationHelper,
+) *AccountRawEvent {
+	return &AccountRawEvent{
+		rdbprojectionbase.NewRDbBase(
+			rdbConn.ToHandle(),
+			"AccountRawEvent",
+		),
+
+		accountAddressPrefix,
+
+		rdbConn,
+		logger,
+
+		migrationHelper,
+	}
+}
+
+func (_ *AccountRawEvent) GetEventsToListen() []string {
+	return []string{
+		event_usecase.BLOCK_RAW_EVENT_CREATED,
+	}
+}
+
+func (projection *AccountRawEvent) OnInit() error {
+	if projection.migrationHelper != nil {
+		projection.migrationHelper.Migrate()
+	}
+	return nil
+}
+
+func (projection *AccountRawEvent) HandleEvents(height int64, events []event_entity.Event) error {
+	var err error
+
+	var rdbTx rdb.Tx
+	rdbTx, err = projection.rdbConn.Begin()
+	if err != nil {
+		return fmt.Errorf("error beginning transaction: %v", err)
+	}
+
+	committed := false
+	defer func() {
+		if !committed {
+			_ = rdbTx.Rollback()
+		}
+	}()
+
+	rdbTxHandle := rdbTx.ToHandle()
+
+	// TODO: Handle genesis transaction
+	if height == int64(0) {
+		if err := projection.UpdateLastHandledEventHeight(rdbTxHandle, height); err != nil {
+			return fmt.Errorf("error updating last handled event height: %v", err)
+		}
+
+		if err := rdbTx.Commit(); err != nil {
+			return fmt.Errorf("error committing changes: %v", err)
+		}
+		committed = true
+		return nil
+	}
+
+	eventsView := NewAccountRawEvents(rdbTxHandle)
+	totalView := NewAccountRawEventsTotal(rdbTxHandle)
+
+	eventRows := make([]view.AccountRawEventRow, 0)
+
+	for _, event := range events {
+		if rawBlockCreatedEvent, ok := event.(*event_usecase.BlockRawEventCreated); ok {
+			accountRawEvent := utils.NewParsedTxsResultLogEvent(&rawBlockCreatedEvent.Data.Content)
+			if rawBlockCreatedEvent.Data.Type == "coin_spent" {
```

Comment:
This list is so magnificent. I have some follow-up questions in mind:

1. Have you read @davcrypto findings on this before https://github.com/crypto-com/chain-indexing/issues/383 ? I think one caveat of relying on event is that some events are not emitted between versions.
2. I wonder if every message handling / chain logic happens emit some kinds of events. Maybe @vincentysc you can share you insight.

Reply:
i think there is no need to switch between `event type`, coz raw event should have no parsing on the event data (other than base64 decode)

Reply:
oh, the switch case is mainly for extracting the related accounts.

Reply:
or is there any better way to extract it?

Reply:
can just assuming the event is related to a specific if there is any event value started with address prefix

Reply:
cool, let me update it then~